### PR TITLE
Data Placement Policy scope

### PR DESCRIPTION
**Pull request by chilung**
in responding to issue #402, draft the first version of data placement scope spec.

### CODE REVIEW

File: extend/data_placement_scope.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+For those data defined in Edge, read/write and exclusive, the data will have an instance in Edge near users. It is also possible to have multiple instances for the same data entity spread among multiple edge sites. We need to define the behavior of data migration according to storage usage and data expiration policy. It will be further defined whether these multiple instances should be synchronized back and saved to Cloud or not.
+
+For those data defined in Edge, read only, the data will have an instance in Edge near users if exclusive. It is also possible to have multiple instances for the same data entity spread among multiple edge sites if shared. We need to define the behavior of data migration according to storage usage and data expiration policy.
+
+In Open Application Model, we can define it as 'DataPlacement' scope. Users can bind 'DataPlacement' to different Components in their application. The implementation of DataPlacemento will handle the data migration detail based on the Scope kind.
```
Comment (hongchaodeng) [inline extend/data_placement_scope.md:13]:
```suggestion
In Open Application Model, we can define it as 'DataPlacement' scope. Users can bind 'DataPlacement' to different Components in their application. The implementation of DataPlacement will handle the data migration detail based on the Scope kind.
```

**Review by hongchaodeng — COMMENTED** [review]
I'm wondering how a user would use this scope? Could you add a section for the user scenario?

File: extend/data_placement_scope.md (hunk 1 above)
Comment (chilung) [reply extend/data_placement_scope.md:13]: I will add a section for scenario and also fix the typo.

Comment (chilung) [reply extend/data_placement_scope.md]: I have added a section for scenario and also fixed the typo.
I also have modified some descriptions, such as 'in-cloud' to replace in Cloud, etc, to keep the explanation and definition in consistence.

File: extend/data_placement_scope.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+In Open Application Model, we can define it as 'DataPlacement' scope. Users can bind 'DataPlacement' to different Components in their application. The implementation of DataPlacement will handle the data migration detail based on the Scope kind.
+
+## Sample User Scenario
+
+In mixed reality application, users usually touch and manipulate the holograms of targeted objects rendered in the vision of real world. The digital twins of the targeted objects can be specified as components bound with 'in-edge' attribute of 'DataPlacement' scope in consideration of the requirement of low latency. The template part of digital twins can be further specified with 'RO' attribute to reflect the fact of templates cannot be modified. On the other hand, the configuration part of digital twins can be specified with 'RW' and 'exclusive' in single user mode. In multi-user mode, The configuration part of digital twins can be specified with 'RW' and 'shared'.
```
Comment (hongchaodeng) [inline extend/data_placement_scope.md:17]: To be honest, I don't understand this scenario. Could you provide a simpler case?

Additionally, I'm also expecting you to go through the scenario with the spec, not just describing the use case.

**Review by hongchaodeng — APPROVED** [review]
LGTM

I will wait a day or two in case other reviewers might have more opinions.

**Comment by samiarahman** [comment]
I am joining this community from data engineering point of view, the data placement scope that you speak of would only apply to components that persist or serve data, how do we ensure the scope does not get forexample applied to the front end component etc.

**Comment by chilung** [comment]
> I am joining this community from data engineering point of view, the data placement scope that you speak of would only apply to components that persist or serve data, how do we ensure the scope does not get forexample applied to the front end component etc.

Could you elaborate this issue in more detail?

**Comment by resouer** [comment]
@chilung For example, for traits we have a `appliestTo` field to say whether this trait applies to certain components or not. This would solve the concern of @samiarahman maybe?

**Comment by samiarahman** [comment]
@resouer  yes that makes sense. Sorry for the late reply.

**Comment by resouer** [comment]
Cool, I will merge this design as a new extended Scope, thanks for contribution!

**Pull request merged**